modules/sfp_peegeepee.py

```diff
@@ -0,0 +1,205 @@
+            return None
+
+        data = re.findall('<li>(.+?)</li>', result_html[0], re.DOTALL)
+
+        if not data:
+            self.sf.debug('No results from PeeGeePee.com')
+            return None
+
+        keys = dict()
+        
+        for key_data in data:
+            fingerprint = re.findall('<code>(.+?)</code>', key_data)
+
+            if not fingerprint:
+                continue
+
+            fingerprint = fingerprint[0].replace(' ', '')
+
+            name_data = ' - '.join(key_data.split(' - ')[1:])
+
+            if not name_data:
+                continue
+
+            description = '/'.join(name_data.split('/')[:-1])
+            email = name_data.split('/')[-1]
+
+            keys[fingerprint] = [description.strip(), email.strip()]
+
+        return keys
+
+    def retrieveKey(self, fingerprint, sourceEvent):
+        res = self.sf.fetchUrl('https://d.peegeepee.com/' + fingerprint + '.asc',
+                               timeout=self.opts['timeout'],
+                               useragent=self.opts['_useragent'])
+
+        time.sleep(1)
+
+        if res['content'] is None:
+            self.sf.debug('No response from d.PeeGeePee.com')
+            return None
+
+        pat = re.compile("(-----BEGIN.*END.*BLOCK-----)", re.MULTILINE | re.DOTALL)
+        matches = re.findall(pat, res['content'])
+
+        for match in matches:
+            self.sf.debug('Found public key: ' + match)
+
+            if len(match) < 300:
+                self.sf.debug('Likely invalid public key.')
+                continue
+
+            evt = SpiderFootEvent('PGP_KEY', match, self.__name__, sourceEvent)
+            self.notifyListeners(evt)
+
+        return None
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+
+        if srcModuleName == 'sfp_peegeepee':
+            return None
+
+        eventData = event.data
+        self.currentEventSrc = event
+
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + ", already checked.")
+            return None
+
+        self.results[eventData] = True
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        keys = self.query(eventData)
+
+        if not keys:
+            self.sf.debug('No results found for ' + eventData)
+            return None
+
+        # Get e-mail addresses on this domain
+        if eventName == 'DOMAIN_NAME' or eventName == 'INTERNET_NAME':
+            for key in keys:
+                email = keys.get(key)[1]
+
+                if not email:
+                    continue
+
+                mailDom = email.lower().split('@')[1]
+
+                if not self.getTarget().matches(mailDom):
+                    continue
+
+                self.sf.info('Found e-mail address: ' + email)
+                evt = SpiderFootEvent('EMAILADDR', email, self.__name__, event)
```

Comment:
In my testing this resulted in duplicates, so dup checking is needed here.

Reply:
Removed duplicates using a temporary `list()` uniqued with `set()`